src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -268,6 +287,63 @@ fn save_directory(
   )
 }
 
+fn materialize_directory(
+  store: Arc<Store>,
+  destination: &Path,
+  fingerprint: &Fingerprint,
+) -> Result<(), ExitError> {
+  let directory = match store.load_directory_proto(&fingerprint)? {
+    Some(d) => d,
+    None => {
+      return Err(ExitError(
+        format!(
+          "Directory with fingerprint {} not found",
+          fingerprint
+        ),
+        ExitCode::NotFound,
+      ));
+    }
+  };
+  make_clean_dir(&destination).map_err(|e| {
+    format!(
+      "Error making directory {:?}: {}",
+      destination,
+      e.description()
+    )
+  })?;
+  for file_node in directory.get_files() {
+    match store.load_file_bytes(&Fingerprint::from_hex_string(
+      &file_node.get_digest().get_hash(),
+    )?)? {
```

Comment:
This much nesting of `?` gets awkward... maybe a local variable?

Reply:
Done